src/details/ArborX_DetailsMortonCode.hpp

```diff
@@ -275,7 +287,7 @@ template <typename Point,
 KOKKOS_INLINE_FUNCTION unsigned int morton32(Point const &p)
 {
   constexpr int DIM = GeometryTraits::dimension_v<Point>;
-  constexpr unsigned N = (1u << (32 / DIM));
+  constexpr auto N = (1lu << (32 / DIM));
```

Comment:
I assume you changed because you wanted at least 64 bits, in which case you need to do `1llu` (`long long unsigned`)